cvat-sdk/cvat_sdk/core/uploading.py

```diff
@@ -89,6 +90,7 @@ def create_url(self):
         headers["upload-length"] = str(self.file_size)
         headers["upload-metadata"] = ",".join(self.encode_metadata())
         resp = self._api_client.rest_client.POST(self.client.url, headers=headers)
+        self.metadata["real_filename"] = resp.headers.get("Upload-Filename")
```

Comment:
IMO, it's not appropriate to use `self.metadata` to store this. `self.metadata` is the stuff that is passed via the `Upload-Metadata` header, which this is not. Just use a custom attribute.